src/doc/style-guide/src/README.md

```diff
@@ -99,6 +99,42 @@ fn bar() {}
 fn baz() {}
 ```
 
+### Sorting
+
+In various cases, the default Rust style specifies to sort things. If not
+otherwise specified, such sorting should be "version sorting", which ensures
+that (for instance) `x8` comes before `x16` even though the character `1` comes
+before the character `8`. (If not otherwise specified, version-sorting is
+lexicographical.)
+
+For the purposes of the Rust style, to compare two strings for version-sorting:
+
+- Compare the strings by (Unicode) character as normal, finding the index of
+  the first differing character. (If the two strings do not have the same
+  length, this may be the end of the shorter string.)
+- For both strings, determine the sequence of ASCII digits containing either
+  that character or the character before. (If either string doesn't have such a
+  sequence of ASCII digits, fall back to comparing the strings as normal.)
+- Compare the numeric values of the number specified by the sequence of digits.
+  (Note that an implementation of this algorithm can easily check this without
+  accumulating copies of the digits or converting to a number: longer sequences
+  of digits are larger numbers, equal-length sequences can be sorted
+  lexicographically.)
```

Comment:
The first sentence and the parenthetical are in contradiction to each other. In general, due to leading zeroes, it is not true that longer sequences correspond to larger values. So the parenthetical, which looks like is just clarifies the intent of the previous sentence, actually fundamentally alters its meaning.

Reply:
The parenthetical wasn't intended to alter meaning, I just forgot to quality "longer sequences of digits (ignoring leading zeroes) are larger numbers".